lib/galaxy/tools/actions/__init__.py

```diff
@@ -903,13 +903,21 @@ def _wrapped_params(
         wrapped_params = WrappedParameters(trans, tool, incoming, input_datasets=input_datasets)
         return wrapped_params
 
-    def _get_on_text(self, inp_data):
+    def _get_on_text(self, inp_data, inp_dataset_collections):
         input_names = []
-        for data in reversed(list(inp_data.values())):
+        collection_names = []
+        for collections in inp_dataset_collections.values():
+            for dataset_collection, _ in collections:
+                if getattr(dataset_collection, "hid", None):
+                    collection_names.append(str(dataset_collection.hid))
```

Comment:
```suggestion
                if isinstance(dataset_collection, HistoryDatasetCollectionAssociation):
                    collection_names.append(str(dataset_collection.hid))
```
? That's the only thing in `inp_dataset_collections` that could have a hid.

Reply:
Guess so, but I do not know.